### ISSUE TITLE
Update Referencing Foreign Keys

### ISSUE THREAD

**Issue by PatrickVienne** [smartninja/wd2-boilerplate#2]
Referencing foreign keys has changed frequently over the last months.

Currently the proper way (as of April 2018) is to define a `ndb.KeyProperty` in the model which can be queried later.

Example:

in `model.py`
```python
class User(ndb.Model):
    nickname = ndb.StringProperty()

class Comments(ndb.Model):
    user = ndb.KeyProperty(kind=User)
    text = ndb.StringProperty()
```

example snippet in `handler.py`
```python
google_user = users.get_current_user()
user = User.query(User.nickname == google_user.nickname()).fetch()
comments = Comments.query(Comments.user == user.key).fetch()
```

**Comment by ramuta** [comment]
KeyProperty has been around for quite some time (way longer than April 2018 which hasn't started yet anyway) :)

The reason why I don't like using it is that it often requires you to do an additional query just to get a key or to construct it manually. Using IntegerProperty is way more useful (or email as a unique identifier in the above example).

**Comment by PatrickVienne** [comment]
ah yes, sorry for the typo ;)

I mean, before ndb, they used the db `ReferenceProperty`.

There are hardly any examples for `KeyProperty`. There is only a short mention in the reference:
https://cloud.google.com/appengine/docs/standard/python/ndb/entity-property-reference

That is why I thought it might be useful to give an example with it.

Of course Email works here, but it invalidates a little bit the SQL classes before, which all talk about using ids as unique foreign keys. The advantage would be that students could easier transfer the example to their own projects, with potentially completely different tables, buy only exchanging the `kind` argument.

**Comment by ramuta** [comment]
Well, it's a NoSQL database, rules are different here :)

As the instructor, you can do it like you proposed in your group (with KeyProperty). But I wouldn't change this boilerplate just because of this.

**Comment by PatrickVienne** [comment]
This is a very good point actually, this datastore thing... ;)

I sticked to the curriculum, but I mentioned it to the students so they can take notes for other projects.
I also would like other trainers to know about it, just in case students ask. I started to put together a `would-be-good-to-know`-list of things a trainer should know how to answer when it comes up in classes. I'll put it up there.
